### PR TITLE
JSDK-2912 | LocalMediaTrack.restart() logs a warning about PeerConnec…

### PR DESCRIPTION
**Pull request by charliesantos**
…tion being closed in p2p Rooms.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

We are keeping references of rtcrtpsender objects even after closing their peerconnections. This PR is to make sure we remove the rtcrtpsender references when their peerconnections are closed.

### CODE REVIEW

File: lib/signaling/v2/peerconnectionmanager.js [hunk 1]
```diff
@@ -188,6 +188,8 @@ class PeerConnectionManager extends QueueingEventEmitter {
       peerConnection.on('stateChanged', function stateChanged(state) {
         if (state === 'closed') {
           peerConnection.removeListener('stateChanged', stateChanged);
+          self._dataTrackSenders.forEach(peerConnection.removeDataTrackSender, peerConnection);
+          self._mediaTrackSenders.forEach(peerConnection.removeMediaTrackSender, peerConnection);
```
Comment (makarandp0) [inline lib/signaling/v2/peerconnectionmanager.js:192]: I think making an arrow function makes it more readable.
```suggestion
          self._dataTrackSenders.forEach(sender => peerConnection.removeDataTrackSender(sender));
          self._mediaTrackSenders.forEach(sender => peerConnection.removeMediaTrackSender(sender));
```

**Review by makarandp0 — APPROVED** [review]
I have a small suggestion, but LGTM otherwise.

**Review by manjeshbhargav — APPROVED** [review]

**Comment by manjeshbhargav** [comment]
@charliesantos Forgot to mention: Please add CHANGELOG entry.

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,6 +19,11 @@ New Features
 
   NOTE: Large Group Rooms is currently in **beta**.
 
+Bug Fixes
+---------
+
+- Fixed a bug where calling `LocalMediaTrack.restart()` logs a warning about PeerConnection being closed in Peer to Peer Rooms. (JSDK-2912)
```
Comment (manjeshbhargav) [inline CHANGELOG.md:25]:
```suggestion
- Fixed a bug where calling `LocalMediaTrack.restart()` logged a warning about PeerConnection being closed in Peer to Peer Rooms. (JSDK-2912)
```

**Review by manjeshbhargav — APPROVED** [review]
@charliesantos 1 nitpick.

**Pull request merged**